lib/chef/provider/package.rb

```diff
@@ -464,10 +464,7 @@ def current_version_array
 
       # @return [Array] new_version(s) as an array
       def new_version_array
-        @new_version_array ||=
-            [ new_resource.version ].flatten.map do |v|
-              ( v.nil? || v.empty? ) ? nil : v
-            end
+        [ new_resource.version ].flatten
```

Comment:
And the other thing you are changing here is the caching of this value in `@new_version_array` because... you say `parse_dependency` somehow _overwrites_ `@new_version_array`? We're pulling a value from it and passing it in. I don't see anywhere where we overwrite `@new_version_array` or `version`. We specifically pass back separate values so we can mutate the arrays here:
https://github.com/ojab/chef/blob/yum_fix/lib/chef/provider/package/yum.rb#L1052

I'm sure I'm just missing something here, but I don't know what it is.